### PR TITLE
python3Packages.readme_renderer: 28.0 -> 29.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 29.0

Change log: https://github.com/pypa/readme_renderer/blob/main/CHANGES.rst#290-2021-02-22

Also, switch to `pytestCheckHook`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114218` at 8ea9fffe run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.mailman-web</li>
    <li>python39Packages.postorius</li>
  </ul>
</details>
<details>
  <summary>46 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/bc9412017f4704e8ad9fe552d752592b">apache-airflow</a></li> (python37Packages.apache-airflow)</li>
    <li><a href="https://gist.github.com/520d74c78d9d5f6e0c1ca20b34debecd">check_systemd</a></li></li>
    <li><a href="https://gist.github.com/4296b1c0df211dd884eaf86e08addbdb">evdevremapkeys</a></li></li>
    <li><a href="https://gist.github.com/92a134115e80bbfc9dccfbeb96f4aafe">keepkey_agent</a></li> (python38Packages.keepkey_agent)</li>
    <li><a href="https://gist.github.com/b8b4ccf5110e9fbcbff316af2c983c79">ledger_agent</a></li> (python38Packages.ledger_agent)</li>
    <li><a href="https://gist.github.com/55a2540a536ac94aa3221c1deec5b63e">luigi</a></li></li>
    <li><a href="https://gist.github.com/47f1c96eff6036e9865f37c5b0bb2f60">mailman-web</a></li> (python38Packages.mailman-web)</li>
    <li><a href="https://gist.github.com/8c7e3b44a54d06433355d271f11cb04b">mnemosyne</a></li></li>
    <li><a href="https://gist.github.com/9426cc7cb20693ef23d5054eb30f42a1">python37Packages.ansible-kernel</a></li></li>
    <li><a href="https://gist.github.com/4cf26a0ade593a30847a548eb89e80a1">python37Packages.ansible-runner</a></li></li>
    <li><a href="https://gist.github.com/dc3379281b9ef570081752814004d5d3">python37Packages.gtts</a></li></li>
    <li><a href="https://gist.github.com/9893966872e061c0f0f647c1e63e7578">python37Packages.keepkey_agent</a></li></li>
    <li><a href="https://gist.github.com/af111fb5aa964ba0ff949df0b23dabb3">python37Packages.labelbox</a></li></li>
    <li><a href="https://gist.github.com/d3f4f120f4f8cb0a5eb08724b80024b9">python37Packages.ledger_agent</a></li></li>
    <li><a href="https://gist.github.com/9639942bfa4316c60b0bc08efe53777d">python37Packages.libagent</a></li></li>
    <li><a href="https://gist.github.com/26354880009119c772ca7fb2f5d409ab">python37Packages.mailman-web</a></li></li>
    <li><a href="https://gist.github.com/36adf5747a252047ba70a3e0219c4199">python37Packages.nagiosplugin</a></li></li>
    <li><a href="https://gist.github.com/52084db7521014840aaca9ab220f802e">python37Packages.ndjson</a></li></li>
    <li><a href="https://gist.github.com/16694064019732e23ba332f777ac2d81">python37Packages.postorius</a></li></li>
    <li><a href="https://gist.github.com/1b357833b10a2c073cadf03d2381a9df">python37Packages.python-daemon</a></li></li>
    <li><a href="https://gist.github.com/875bf59d0746ddba9dab42228c297c11">python37Packages.readme</a></li></li>
    <li><a href="https://gist.github.com/48223f8cf132ab062be70bfce745042e">python37Packages.readme_renderer</a></li></li>
    <li><a href="https://gist.github.com/59f665e21ddecf6910f9aadf4f39d377">python37Packages.restview</a></li></li>
    <li><a href="https://gist.github.com/f1e9de58cafe0616d7cfd885ea3e3b90">python37Packages.salmon-mail</a></li></li>
    <li><a href="https://gist.github.com/e715ab5c0c40c25237f6e5f90e612d63">python37Packages.toggl-cli</a></li></li>
    <li><a href="https://gist.github.com/2e832ddaf2cfd83b6c509e8a9631d3d9">python37Packages.trezor_agent</a></li></li>
    <li><a href="https://gist.github.com/06deb2feae87abdab836a087367e25a0">python37Packages.twine</a></li></li>
    <li><a href="https://gist.github.com/2c2fcddf85bb6e8ccf6232141a2381a1">python38Packages.ansible-kernel</a></li></li>
    <li><a href="https://gist.github.com/a80d4c6ec78ecb66cf9fb48bc0ad878b">python38Packages.ansible-runner</a></li></li>
    <li><a href="https://gist.github.com/2bc905f9fdbe473c63b4f137c97f61fc">python38Packages.gtts</a></li></li>
    <li><a href="https://gist.github.com/0c20cc5f211574df7e1bd112c2fa1d4d">python38Packages.labelbox</a></li></li>
    <li><a href="https://gist.github.com/2f77b04c25a40e1bd2ad471996243ab0">python38Packages.libagent</a></li></li>
    <li><a href="https://gist.github.com/5e25bba1b55a3cf0baf7293902d77628">python38Packages.nagiosplugin</a></li></li>
    <li><a href="https://gist.github.com/c3c0c0f644b4603bb42cea48fd3a9044">python38Packages.ndjson</a></li></li>
    <li><a href="https://gist.github.com/8560ffdfaeaf8faa4633f36851d7bd46">python38Packages.postorius</a></li></li>
    <li><a href="https://gist.github.com/c8b428d726a477309e0004ac1f448777">python38Packages.python-daemon</a></li></li>
    <li><a href="https://gist.github.com/4a33a4e42d8f3df560858d8c9b0f0938">python38Packages.readme</a></li></li>
    <li><a href="https://gist.github.com/42de579ced89773577bedd8631fbf93c">python38Packages.readme_renderer</a></li></li>
    <li><a href="https://gist.github.com/becdcced0bfde022f7510a5ad48a8e12">python38Packages.restview</a></li></li>
    <li><a href="https://gist.github.com/e3ddd5d6bfcc9fa6bfbd2d344b8b4099">python38Packages.salmon-mail</a></li></li>
    <li><a href="https://gist.github.com/8b7c8ab9deef591be08558fd99efdbf6">python38Packages.toggl-cli</a></li></li>
    <li><a href="https://gist.github.com/7947e1c6423194ab153f8bcb9f061ef3">trezor_agent</a></li> (python38Packages.trezor_agent)</li>
    <li><a href="https://gist.github.com/f8a46d8e733db5226cc68e7db2759045">python38Packages.twine</a></li></li>
    <li><a href="https://gist.github.com/8538bf67f3f09d269899af16464b8c1b">python39Packages.readme</a></li></li>
    <li><a href="https://gist.github.com/e7afff28365ba125f83d7b2c03911443">python39Packages.readme_renderer</a></li></li>
    <li><a href="https://gist.github.com/b460250f264fe61073685367342d8c5d">python39Packages.restview</a></li></li>
  </ul>
</details>

**Comment by dotlambda** [comment]
> Drop support for Python 2 and 3.5 (#188)

That should be reflected in the expression.

**Comment by fabaff** [comment]
> That should be reflected in the expression.

Indeed, that should be added. Thanks for the feedback.

**Comment by fabaff** [comment]
Depends on #112854 for Python 3.9.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114218` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python39Packages.django-mailman3</li>
    <li>python39Packages.hyperkitty</li>
    <li>python39Packages.mailman-web</li>
    <li>python39Packages.postorius</li>
  </ul>
</details>
<details>
  <summary>50 packages built:</summary>
  <ul>
    <li>apache-airflow (python37Packages.apache-airflow)</li>
    <li>check_systemd</li>
    <li>evdevremapkeys</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>luigi</li>
    <li>mailman-web (python38Packages.mailman-web)</li>
    <li>mnemosyne</li>
    <li>python37Packages.ansible-kernel</li>
    <li>python37Packages.ansible-runner</li>
    <li>python37Packages.django-mailman3</li>
    <li>python37Packages.gtts</li>
    <li>python37Packages.hyperkitty</li>
    <li>python37Packages.keepkey_agent</li>
    <li>python37Packages.labelbox</li>
    <li>python37Packages.ledger_agent</li>
    <li>python37Packages.libagent</li>
    <li>python37Packages.mailman-web</li>
    <li>python37Packages.nagiosplugin</li>
    <li>python37Packages.ndjson</li>
    <li>python37Packages.postorius</li>
    <li>python37Packages.python-daemon</li>
    <li>python37Packages.readme</li>
    <li>python37Packages.readme_renderer</li>
    <li>python37Packages.restview</li>
    <li>python37Packages.salmon-mail</li>
    <li>python37Packages.toggl-cli</li>
    <li>python37Packages.trezor_agent</li>
    <li>python37Packages.twine</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.ansible-runner</li>
    <li>python38Packages.django-mailman3</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.hyperkitty</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.libagent</li>
    <li>python38Packages.nagiosplugin</li>
    <li>python38Packages.ndjson</li>
    <li>python38Packages.postorius</li>
    <li>python38Packages.python-daemon</li>
    <li>python38Packages.readme</li>
    <li>python38Packages.readme_renderer</li>
    <li>python38Packages.restview</li>
    <li>python38Packages.salmon-mail</li>
    <li>python38Packages.toggl-cli</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>python38Packages.twine</li>
    <li>python39Packages.readme</li>
    <li>python39Packages.readme_renderer</li>
    <li>python39Packages.restview</li>
  </ul>
</details>

File: pkgs/development/python-modules/readme_renderer/default.nix [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
-, six
+, pytestCheckHook
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "readme_renderer";
-  version = "28.0";
+  version = "29.0";
+  disabled = pythonOlder "3.5";
```
Comment (dotlambda) [inline pkgs/development/python-modules/readme_renderer/default.nix:17]:
```suggestion
  disabled = pythonOlder "3.6";
```
Support for 3.5 was dropped.

**Review by dotlambda — DISMISSED** [review]

File: pkgs/development/python-modules/readme_renderer/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/readme_renderer/default.nix]: Ähh, damn copy-&-paste...

**Pull request merged**